scripts/check_no_std.sh

```diff
@@ -0,0 +1,26 @@
+#!/usr/bin/env bash
+set -eo pipefail
+
+target=riscv32imac-unknown-none-elf
+crates=(
+    alloy-core
+    alloy-core-sol-test
+    alloy-dyn-abi
+    alloy-json-abi
+    alloy-json-abi
+    alloy-primitives
+    # alloy-sol-macro
+    # alloy-sol-macro-expander
+    # alloy-sol-macro-input
+    alloy-sol-type-parser
+    alloy-sol-types
+    # syn-solidity
+)
+
+cmd=(cargo +stable hack check --no-default-features --target "$target")
```

Comment:
I think we'll either need to use `+nightly` here or bump the rust-version in the cargo toml to `1.81` since the core Error trait is stabilized.

Reply:
I can just disable the error feature